### PR TITLE
Add support for memory-profiling on subsystem-bench

### PR DESCRIPTION
**Pull request by alexggh**
Add support in subsystem-benchmarks to profile memory usage using the jemalloc builting profiler, this allows us to run each benchmark with profiling enabled and determine if the memory usage patters are in conformance with our expectations.

### CODE REVIEW

File: polkadot/node/subsystem-bench/src/cli/subsystem-bench.rs [hunk 1]
```diff
@@ -182,6 +182,17 @@ impl BenchCli {
...
+#[cfg(feature = "memprofile")]
+#[global_allocator]
+static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
+#[cfg(feature = "memprofile")]
+#[allow(non_upper_case_globals)]
+#[export_name = "malloc_conf"]
+// See https://jemalloc.net/jemalloc.3.html for more information on the configuration options.
+pub static malloc_conf: &[u8] =
+	b"prof:true,prof_active:true,lg_prof_interval:30,lg_prof_sample:21,prof_prefix:/tmp/subsystem-bench\0";
```
Comment (ordian) [inline polkadot/node/subsystem-bench/src/cli/subsystem-bench.rs:194]: I've use some of these options some time ago: https://gist.github.com/ordian/928dc2bd45022cddd547528f64db9174#gistcomment-4807555
but configured them via an env variable. This looks nice :)

File: polkadot/node/subsystem-bench/README.md [hunk 2]
```diff
@@ -260,6 +260,32 @@ This file is best interpreted with `cg_annotate --auto=yes cachegrind.out.<pid>`
 
 For finer profiling of cache misses, better use `perf` on a bare-metal machine.
 
+### Profile memory usage using jemalloc
```
[the remark this answers is not in the document]
Comment (alexggh) [reply polkadot/node/subsystem-bench/README.md:263]: I've never used it, but looks nice, I will add it to the readme.

**Review by tdimitrov — APPROVED** [review]

File: polkadot/node/subsystem-bench/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexggh) [reply polkadot/node/subsystem-bench/README.md:263]: Done.

File: polkadot/node/subsystem-bench/README.md [hunk 3]
```diff
@@ -260,6 +260,35 @@ This file is best interpreted with `cg_annotate --auto=yes cachegrind.out.<pid>`
...
+
+Bellow you can find instructions how to setup and run profiling with jemalloc, this is complementary with using other memory profiling tools
+like: https://github.com/koute/bytehound?tab=readme-ov-file#basic-usage.
+
+#### Prerequisites
+Install tooling with:
+```
+sudo apt install libjemalloc-dev graphviz
```
Comment (sandreim) [inline polkadot/node/subsystem-bench/README.md:271]: nit: I guess this only works for some Linux distributions, but I am a mac user :)

**Review by sandreim — COMMENTED** [review]
thanks @alexggh

**Review by sandreim — APPROVED** [review]

File: polkadot/node/subsystem-bench/README.md (hunk 3 above)
Comment (alexggh) [reply polkadot/node/subsystem-bench/README.md:271]: Yes, linux only :D

**Review by alindima — APPROVED** [review]

**Pull request merged**